packages/drivers/odsp-driver/src/odspDeltaStorageService.ts

```diff
@@ -43,69 +44,77 @@ export class OdspDeltaStorageService {
 		telemetryProps: ITelemetryProperties,
 		scenarioName?: string,
 	): Promise<IDeltasFetchResult> {
-		return getWithRetryForTokenRefresh(async (options) => {
-			// Note - this call ends up in getSocketStorageDiscovery() and can refresh token
-			// Thus it needs to be done before we call getStorageToken() to reduce extra calls
-			const baseUrl = this.buildUrl(from, to);
-			const storageToken = await this.getStorageToken(options, "DeltaStorage");
-
-			const formBoundary = uuid();
-			let postBody = `--${formBoundary}\r\n`;
-			postBody += `Authorization: Bearer ${storageToken}\r\n`;
-			postBody += `X-HTTP-Method-Override: GET\r\n`;
-
-			postBody += `_post: 1\r\n`;
-			postBody += `\r\n--${formBoundary}--`;
-			const headers: { [index: string]: any } = {
-				"Content-Type": `multipart/form-data;boundary=${formBoundary}`,
-			};
-
-			// Some request take a long time (1-2 minutes) to complete, where telemetry shows very small amount
-			// of time spent on server, and usually small payload sizes. I.e. all the time is spent somewhere in
-			// networking. Even bigger problem - a lot of requests timeout (based on cursory look - after 1-2 minutes)
-			// So adding some timeout to ensure we retry again in hope of faster success.
-			// Please see https://github.com/microsoft/FluidFramework/issues/6997 for details.
-			const abort = new AbortController();
-			const timer = setTimeout(() => abort.abort(), 30000);
-
-			const response = await this.epochTracker.fetchAndParseAsJSON<IDeltaStorageGetResponse>(
-				baseUrl,
-				{
-					headers,
-					body: postBody,
-					method: "POST",
-					signal: abort.signal,
-				},
-				"ops",
-				true,
-				scenarioName,
-			);
-			clearTimeout(timer);
-			const deltaStorageResponse = response.content;
-			const messages =
-				deltaStorageResponse.value.length > 0 && "op" in deltaStorageResponse.value[0]
-					? (deltaStorageResponse.value as ISequencedDeltaOpMessage[]).map(
-							(operation) => operation.op,
-					  )
-					: (deltaStorageResponse.value as ISequencedDocumentMessage[]);
-
-			this.logger.sendPerformanceEvent({
+		return PerformanceEvent.timedExecAsync(
```

Comment:
I'd move it below getWithRetryForTokenRefresh() call, as it already reports an event with how long it takes, so it's useful to have events with durations that we can easily sum up.

Reply:
With my current change, getToken time will be reported separately from the OPsFetch time. Previously we were reporting OPsFetch only in successful case and not in failure cases.